### PR TITLE
Eliminate data races

### PR DESCRIPTION
**Pull request by mxgrey**
As reported in https://github.com/open-rmf/rmf_ros2/issues/82, Thread Sanitizer is complaining about many data races that are occurring in the `rmf_fleet_adapter` test code.

This PR combined with https://github.com/open-rmf/rmf_traffic/pull/46 should eliminate the vast majority (possibly all?) of them. After updating to these changes I have not seen a data race reported on the tests in a few dozen runs.

### CODE REVIEW

**Comment by mxgrey** [comment]
Note that CI won't be able to pass until open-rmf/rmf_traffic#46 is merged.

**Review by aaronchongth — APPROVED** [review]
Thanks for tracking down the data races and fixing them! :bow: Just need to update the base branch, and we are good to go!

**Comment by aaronchongth** [comment]
The failures seem to be coming from this possibly stale or incomplete test, https://github.com/open-rmf/rmf_ros2/blob/main/rmf_fleet_adapter/rmf_rxcpp/test/test_Transport.cpp#L104-L108 :thinking: And it looks like it was possibly to the section name clashing? I removed it and the data races were resolved.

File: rmf_fleet_adapter/test/phases/test_GoToPlace.cpp [hunk 1]
```diff
@@ -187,10 +187,14 @@ SCENARIO_METHOD(MockAdapterFixture, "go to place negotiation", "[phases]")
   const auto context_1 = info_1.context;
 
   const auto pending_0 = phases::GoToPlace::make(context_0, {start_0}, goal_0);
-  const auto active_0 = pending_0->begin();
+  const auto active_0 = info_0.schedule_and_wait(
+    std::function<std::shared_ptr<Task::ActivePhase>()>(
+      [&](){ return pending_0->begin(); }));
```
Comment (aaronchongth) [inline rmf_fleet_adapter/test/phases/test_GoToPlace.cpp:192]:
```suggestion
      [&]() { return pending_0->begin(); }));
```

File: rmf_fleet_adapter/test/phases/test_GoToPlace.cpp [hunk 2]
```diff
@@ -187,10 +187,14 @@ SCENARIO_METHOD(MockAdapterFixture, "go to place negotiation", "[phases]")
...
 
   const auto pending_1 = phases::GoToPlace::make(context_1, {start_1}, goal_1);
-  const auto active_1 = pending_1->begin();
+  const auto active_1 = info_1.schedule_and_wait(
+    std::function<std::shared_ptr<Task::ActivePhase>()>(
+      [&](){ return pending_1->begin(); }));
```
Comment (aaronchongth) [inline rmf_fleet_adapter/test/phases/test_GoToPlace.cpp:197]:
```suggestion
      [&]() { return pending_1->begin(); }));
```

**Review by aaronchongth — CHANGES_REQUESTED** [review]
Looks like the final failures are just linting issues, it seems to be complaining about these lines not indented, https://github.com/open-rmf/rmf_ros2/pull/129/files#diff-a5d4dd01175bb6932faad974abcee14d99fff47df3dc28e444ac5401594a951cR312-R333, as well as the bottom few suggestions I made

**Comment by mxgrey** [comment]
@aaronchongth All checks are passing now. We just need a re-approval.

**Review by aaronchongth — APPROVED** [review]
LGTM!

**Pull request merged**